### PR TITLE
GCN candidate

### PR DESCRIPTION
**Pull request by ezzarghili**
Graal Cloud Native GCN is an open source project (https://github.com/oracle/gcn) for cloud-native microservices.
First version we are submitting as PR, we will follow the vendor API process for future releases.

### CODE REVIEW

**Comment by ezzarghili** [comment]
First release srtifacts are published now and can be now consumed by users using sdkman when these change are merged.

File: src/main/scala/io/sdkman/changelogs/GcnMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package io.sdkman.changelogs
+
+import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.mongodb.client.MongoDatabase
+
+@ChangeLog(order = "080")
+class MicronautMigrations {
```
Comment (burtbeckwith) [inline src/main/scala/io/sdkman/changelogs/GcnMigrations.scala:7]: shouldn't this be called `GcnMigrations`?

**Comment by ezzarghili** [comment]
> shouldn't this be called `GcnMigrations`? 

Indeed that was a missed change from using Micronaut as reference.

File: src/main/scala/io/sdkman/changelogs/GcnMigrations.scala (hunk 1 above)
Comment (ezzarghili) [reply src/main/scala/io/sdkman/changelogs/GcnMigrations.scala]: Indeed that was a missed change from using Micronaut as reference.

**Comment by marc0der** [comment]
All our Java and Graal versions are published by an automated process consuming the Disco API. Perhaps we can leverage this instead? /cc @eddumelendez

**Comment by ezzarghili** [comment]
For GraalVM releases, once we onboard the vendor API we can also take over publishing new releases including non community ones that are not yet available on SDKMAN

For this PR, it is not about GraalVM, people will still need to install GraalVM separately, this change is for CLI tool of GCN that allow people to generate projects and interact with their GCN codebase.

File: src/main/scala/io/sdkman/changelogs/GcnMigrations.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package io.sdkman.changelogs
+
+import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.mongodb.client.MongoDatabase
+
+@ChangeLog(order = "080")
```
Comment (eddumelendez) [inline src/main/scala/io/sdkman/changelogs/GcnMigrations.scala:6]: There is an error in other migration and `079` is duplicated. So, one of those should move to `080`.

```suggestion
@ChangeLog(order = "081")
```

**Review by eddumelendez — CHANGES_REQUESTED** [review]
Thanks for the PR @ezzarghili ! Can you please perform `./gradlew scalafmt`?

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
After running `sdk update` the candidate is listed `sdk list gcn`

**Comment by ezzarghili** [comment]
@eddumelendez I based my changes on existing code that suggest `tag.gz` are used too (e.g src/main/scala/io/sdkman/changelogs/JdkMissionControlMigrations.scala)

Testing now it seems that we are required to use zip archives?
image

Should we move this to use zip for all platforms?

**Comment by marc0der** [comment]
@ezzarghili yes, we only allow tarballs in special cases for now. Some work is on the way to allow both, but is not ready yet. For now zips are preferable. Will you also be using our vendor API? If so, you can use the PATCH method to change versions without needing database migrations.

**Comment by ezzarghili** [comment]
Thank you @marc0der, I sent my public key for the vendor API access credentails, will use the API to fix this.

**Comment by ezzarghili** [comment]
@marc0der any idea when to expect the access credentials? thanks

**Pull request merged**